src/HttpClientInterception/HttpRequestInterceptionBuilder.cs

```diff
@@ -111,26 +111,46 @@ public HttpRequestInterceptionBuilder For(Predicate<HttpRequestMessage>? predica
     /// Pass a value of <see langword="null"/> to remove a previously-registered custom request matching predicate.
     /// </remarks>
     public HttpRequestInterceptionBuilder For(params Predicate<HttpRequestMessage>[] predicates)
+    {
+        return For(predicates
+            .Select(predicate => new Func<HttpRequestMessage, Task<bool>>((message) => Task.FromResult(predicate(message))))
+            .ToArray());
```

Comment:
This needs to null check independently, rather than call the other method. Otherwise you're just going to end up with exceptions thrown if this method is passed null.

See https://github.com/justeat/httpclient-interception/blob/570042f6adbaba7d7c41201620c70c838ce6ecc4/src/HttpClientInterception/HttpRequestInterceptionBuilder.cs#L800-L884 for examples where `DelegateHelpers` is used to remove the need to manually change the delegate signatures and null check things.